wdl/conditional-analysis/regenie_conditional_full.wdl

```diff
@@ -0,0 +1,423 @@
+version 1.0
+
+workflow conditional_analysis {
+
+  input {
+    String docker
+    File phenos_to_cond
+    String release
+    
+    String mlogp_col
+    String chr_col
+    String pos_col
+    String ref_col
+    String alt_col
+    Float conditioning_mlogp_threshold
+    Float locus_mlogp_threshold  
+    Array[String] chroms
+    Array[String] covariates
+    File pheno_file
+    String sumstats_root
+
+  }
+
+  String prefix = "finngen_R" + release
+  # returns covariate string for each pheno
+  call filter_covariates {input: docker=docker,pheno_file=pheno_file,pheno_list = phenos_to_cond,covariates=covariates}  
+  Array[String] pheno_data = read_lines(phenos_to_cond)
+  # go through phenos
+  scatter (p in pheno_data) {
+    #get hits under pval threshold
+    call extract_cond_regions {
+      input: pheno=p, mlogp_threshold = locus_mlogp_threshold, docker=docker,mlogp_col = mlogp_col,chr_col=chr_col,pos_col = pos_col,ref_col=ref_col,alt_col=alt_col,chroms=chroms,sumstats_root=sumstats_root
+    }
+   }
+
+   #SINGLE JOB PER LOCUS VERSION
+   call merge_regions {input: docker =docker,hits=extract_cond_regions.gw_sig_res}
+
+   Map[String,String] cov_map = read_map(filter_covariates.cov_pheno_map)
+   Array[Array[String]] all_regions = read_tsv(merge_regions.regions)
+   
+   # loop over all regions running one variant per shard
+   scatter (region in all_regions) {
```

Comment:
Do you really need the extract_cond_regions and merge_regions at all?
If your input would be bed file containing the regions and column for PHENO then you could loop over that file and in each regenie_conditional task you'd pull the top variant in the region to start conditioning on?

Reply:
Well kinda. Based on the pval filter the regions can be empty, so you'd have machines firing up for potentially no good reason. 
I guess i copied the original wdl (there was a nested scatter iirc) and there might be some other practical reasons I can't remember (there was also the one machine per pheno/chrom combination where i could use the same logic also iirc).

I don't think it should make a big difference in the overall costs though, the two tasks take minutes to run.

Reply:
also if you would first scatter per pheno and call sub workflow that would do conditional per pheno, it would be extremely easy and convenient to merge the per phenotype results within subworkflow without inferring phenos from file names

Reply:
> we find the 'no significant things to process' in the analysis machine and in our case we always have results. "I don't think it should make a big difference in the overall costs though, the two tasks take minutes to run." Not for running costs but maintenance is always easier (especially if someone else is extending than original author which is often the case) when software is not overly complicated.

That makes sense, but it also made sense at the time, while debugging not to have two conceptually separate jobs run, when the output of the first one is not subject to change and while the other (the regenie conditional) took several number of iterations to nail down. 

if you want i can rehaul the pipeline, but it will definitely come at a cost in terms of time.

Reply:
I think it's worth simplifying the pipeline so go ahead and modify to subworkflow approach.....